src/AggregateFunctions/AggregateFunctionSumMap.cpp

```diff
@@ -52,23 +52,37 @@ using SumMapArgs = std::pair<DataTypePtr, DataTypes>;
 
 SumMapArgs parseArguments(const std::string & name, const DataTypes & arguments)
 {
-    if (arguments.size() < 2)
-        throw Exception("Aggregate function " + name + " requires at least two arguments of Array type.",
+    DataTypes args;
+
+    if (arguments.size() == 1)
+    {
+        const auto * tuple_type = checkAndGetDataType<DataTypeTuple>(arguments[0].get());
```

Comment:
```suggestion
        // sumMap is a transitive function, so it can be stored in SimpleAggregateFunction columns.
        // There is a caveat: it must support sumMap(sumMap(...)), e.g. it must be able to accept its
        // own output as an input. This is why we also support Tuple(keys, values) as an argument. 
        const auto * tuple_type = checkAndGetDataType<DataTypeTuple>(arguments[0].get());
```
I think we should mention why we support tuples, maybe even in the docs. Does this comment make sense?

Reply:
Yes, thanks, it makes sense.